[lte][agw] Disable smsd AGW service by default

shaddi opened the pull request:
Signed-off-by: Shaddi Hasan <shasan@fb.com>

## Summary

~Disables smsd AGW service by default, and adds a mechanism to enable/disable services via configuration. When a service is disabled, its event loop is not started, and `run()` does nothing.~

Based on feedback, disables smsd AGW service unless the MME is in `SMS_ORC8R` mode. 

## Test Plan

CI + testing on VM

## Additional Information

shaddi commented:
Bump @hcgatewood @andreilee @mpgermano

ulaskozat reviewed (changes requested):
Can you tie this configuration to the config on MME instead?

ardzoht commented:
Should we make smsd a dynamic service instead? This way the service can be enabled / disabled from mconfig and furthermore, cloud config.

shaddi committed:
Run smsd AGW service only if MME in SMS_ORC8R mode

Check the mme mconfig to determine the non_eps_service_control parameter
for the MME, and only run the AGW SMS relay if the service is enabled in
the MME.

Signed-off-by: Shaddi Hasan <shaddih@gmail.com>

ulaskozat reviewed (approved):
LGTM

pshelar commented on lte/gateway/python/magma/smsd/main.py:
```diff
@@ -43,5 +46,23 @@ def main():
+        elif non_eps_service_control == 3:
+            return "SMS_ORC8R"
+    return "OFF"
+
```
this keeps restarting the service. is it expected?
```
vagrant@magma-dev:~/magma/lte/gateway$ sudo service magma@smsd status
● magma@smsd.service - Magma smsd service
   Loaded: loaded (/etc/systemd/system/magma@.service; disabled; vendor preset: enabled)
   Active: activating (auto-restart) since Thu 2021-03-04 16:46:17 UTC; 3s ago
  Process: 7145 ExecStopPost=/usr/bin/env python3 /usr/local/bin/service_util.py smsd (code=exited, status=0/SUCCESS)
  Process: 7137 ExecStart=/usr/bin/env python3 -m magma.smsd.main (code=exited, status=0/SUCCESS)
 Main PID: 7137 (code=exited, status=0/SUCCESS)
vagrant@magma-dev:~/magma/lte/gateway$ sudo service magma@smsd status
● magma@smsd.service - Magma smsd service
   Loaded: loaded (/etc/systemd/system/magma@.service; disabled; vendor preset: enabled)
   Active: activating (auto-restart) since Thu 2021-03-04 16:46:23 UTC; 2s ago
  Process: 7197 ExecStopPost=/usr/bin/env python3 /usr/local/bin/service_util.py smsd (code=exited, status=0/SUCCESS)
  Process: 7189 ExecStart=/usr/bin/env python3 -m magma.smsd.main (code=exited, status=0/SUCCESS)
 Main PID: 7189 (code=exited, status=0/SUCCESS)
vagrant@magma-dev:~/magma/lte/gateway$ sudo service magma@smsd status
● magma@smsd.service - Magma smsd service
   Loaded: loaded (/etc/systemd/system/magma@.service; disabled; vendor preset: enabled)
   Active: activating (auto-restart) since Thu 2021-03-04 16:46:28 UTC; 2s ago
  Process: 7218 ExecStopPost=/usr/bin/env python3 /usr/local/bin/service_util.py smsd (code=exited, status=0/SUCCESS)
  Process: 7209 ExecStart=/usr/bin/env python3 -m magma.smsd.main (code=exited, status=0/SUCCESS)
 Main PID: 7209 (code=exited, status=0/SUCCESS)
```

shaddi replied:
Yes. The service terminates if the MME config is not in the SMS_ORC8R mode and is restarted by systemd. This way, if the MME config is changed from the orc8r, the smsd relay will start in the right mode automatically. If we're concerned about log noise, we can do this config check loop in the service; this approach just seemed cleaner to me.

pshelar replied:
Yes, I think we can go with checking config in the service.
I am bit concerned about ERROR logging.
```
Mar  4 17:49:35 magma-dev magmad[4799]: ERROR:root:GetServiceInfo Error for smsd! [StatusCode.UNAVAILABLE] failed to connect to all addresses
```

shaddi commented:
Updated to address @pshelar's comment. The service will start, which allows `magmad` to get service info normally (eliminating the error message), but we'll no-op when the MME is not in `SMS_ORC8R` mode to avoid unnecessary polling of the orc8r.

pshelar reviewed (approved):
Thanks for the fix.
